Pod/Classes/ResultType.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ResultType.swift
+//  Pods
+//
+//  Created by Bueno McCartney on 12/7/16.
+//
+//
+
+import Foundation
+
+public protocol ResultType {
+    associatedtype Value
+    
+    init(_ value: Value?)
```

Comment:
Why is `Value` taking an optional type? Since Value itself is a generic, shouldn’t passing an optional work without the `?`?